docs/reference/builder.md

```diff
@@ -203,43 +210,34 @@ The following is an example `.dockerignore` file:
 
 | Rule           | Behavior                                                                                                                                                                     |
 |----------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
-| `*/temp*`      | Exclude all files with names starting with`temp` in any subdirectory below the root directory. For example, a file named`/somedir/temporary.txt` is ignored.                 |
+| `*/temp*`      | Exclude all files with names starting with`temp` in any subdirectory immediately below the root directory. For example, a plain file named`/somedir/temporary.txt` is ignored, as is a directory named `/somedir/temp`.                 |
 | `*/*/temp*`    | Exclude files starting with name `temp` from any subdirectory that is two levels below the root directory. For example, the file `/somedir/subdir/temporary.txt` is ignored. |
 | `temp?`        | Exclude the files that match the pattern in the root directory. For example, the files `tempa`, `tempb` in the root directory are ignored.                                   |
-| `*.md `        | Exclude all markdown files in the root directory.                                                                                                                            |
-| `!LICENSE.md`  | Exception to the Markdown files exclusion is this file,  `LICENSE.md`, Include this file in the build.                                                                       |
-
-The placement of  `!` exception rules influences the matching algorithm; the
-last line of the `.dockerignore` that matches a particular file determines
-whether it is included or excluded. In the above example, the `LICENSE.md` file
-matches both the  `*.md` and `!LICENSE.md` rule. If you reverse the lines in the
-example:
-
-```
-    */temp*
-    */*/temp*
-    temp?
-    !LICENSE.md
-    *.md
-```
-
-The build would exclude `LICENSE.md` because the last `*.md` rule adds all
-Markdown files in the root directory back onto the ignore list. The
-`!LICENSE.md` rule has no effect because the subsequent `*.md` rule overrides
-it.
-
-You can even use the  `.dockerignore` file to ignore the `Dockerfile` and
-`.dockerignore` files. This is useful if you are copying files from the root of
-the build context into your new container but do not want to include the
-`Dockerfile` or `.dockerignore` files (e.g. `ADD . /someDir/`).
-
+| `*.md `        | Exclude all markdown files in the root directory...       |                                  
+| `!README*.md`  | ... except those matching `README*.md`. These markdown file *will* be included in the build...
+| `README-secret.md`    | ... except `README-secret.md`.  This file *won't* be included in the build. |
+
+
+Sometimes, you may want to specify which files to include in the build
```

Comment:
Not sure why you removed the section starting "the placement" ...  Your "anit-ignore" file example is a nice addition but the concept should come last because it is the antithesis of what has gone before, so...

---> https://gist.github.com/moxiegirl/4c12fe428e709ba5407c

The placement of  `!` exception rules influences the matching algorithm; the
last line of the `.dockerignore` that matches a particular file determines
whether it is included or excluded. In the above example, the `LICENSE.md` file
matches both the  `*.md` and `!LICENSE.md` rule. If you reverse the lines in the
example:

```
    */temp*
    */*/temp*
    temp?
    !LICENSE.md
    *.md
```

The build would exclude `LICENSE.md` because the last `*.md` rule adds all
Markdown files in the root directory back onto the ignore list. The
`!LICENSE.md` rule has no effect because the subsequent `*.md` rule overrides
it.

You can even use the  `.dockerignore` file to ignore the `Dockerfile` and
`.dockerignore` files. This is useful if you are copying files from the root of
the build context into your new container but do not want to include the
`Dockerfile` or `.dockerignore` files (e.g. `ADD . /someDir/`).

Finally, you may want to specify which files to include in the build
context, rather than which to ignore. To achieve this, specify `*` as the first pattern, followed by one or more  `!` exclusion patterns.  

**Note**: For historical reasons, the pattern `.` is always ignored.